### PR TITLE
refactor: hide `RegionRoute` behind `TableRouteValue`

### PR DESCRIPTION
**Pull request by MichaelScofield**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

Metric Engine table route, part 2

Next PR will create metric engine table, with its table route.

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

## Refer to a related PR or issue link (optional)

#2864

### CODE REVIEW

File: src/common/meta/src/ddl/create_table.rs [hunk 1]
```diff
@@ -329,29 +364,31 @@ impl TableCreator {
...
                 task,
-                region_routes,
+                table_route,
                 region_wal_options,
             },
             opening_regions: vec![],
         }
     }
 
     /// Register opening regions if doesn't exist.
```
Comment (WenyXu) [inline src/common/meta/src/ddl/create_table.rs:382]:
```suggestion
    /// Registers and returns the guards of the opening region if they don't exist.
```

File: src/common/meta/src/key.rs [hunk 2]
```diff
@@ -403,30 +400,28 @@ impl TableMetadataManager {
...
+            let create_datanode_table_txn = self.datanode_table_manager().build_create_txn(
+                table_id,
+                &engine,
+                &region_storage_path,
+                region_options,
+                region_wal_options,
+                region_distribution(&x.region_routes)?,
+            )?;
+            txn = txn.merge(create_datanode_table_txn);
+        }
```
Comment (WenyXu) [inline src/common/meta/src/key.rs:423]: Should we register the logical table route for the metric table? @waynexia Maybe we need to add a TODO here.

File: src/meta-srv/src/procedure/region_migration/test_util.rs [hunk 3]
```diff
@@ -144,6 +145,21 @@ impl TestingEnv {
...
+    ) {
+        self.table_metadata_manager
+            .create_table_metadata(
+                table_info,
+                TableRouteValue::new_physical(region_routes),
+                HashMap::default(),
+            )
+            .await
+            .unwrap();
+    }
```
Comment (WenyXu) [inline src/meta-srv/src/procedure/region_migration/test_util.rs:162]:
```suggestion
    // Creates a table with the physical table route. 
    pub async fn create_physical_table_metadata(
        &self,
        table_info: RawTableInfo,
        region_routes: Vec<RegionRoute>,
    ) {
        self.table_metadata_manager
            .create_table_metadata(
                table_info,
                TableRouteValue::new_physical(region_routes),
                HashMap::default(),
            )
            .await
            .unwrap();
    }
```

File: src/common/meta/src/key.rs (hunk 2 above)
Comment (MichaelScofield) [reply src/common/meta/src/key.rs:423]: Do logical regions need to open explicitly on datanodes? I think it's better not to, make them opened already by opening the physical regions. This way we can truly untie the relationship between datanode and logical region.

Comment (WenyXu) [reply src/common/meta/src/key.rs:423]: Oh, my mistake. I mixed the `DatanodeTableValue` and `TableRouteValue`.

**Comment by MichaelScofield** [comment]
@WenyXu PTAL

**Comment by WenyXu** [comment]
> @WenyXu PTAL

Rustfmt failed.

**Comment by MichaelScofield** [comment]
@WenyXu fixed

**Review by WenyXu — APPROVED** [review]
LGTM

File: src/common/meta/src/key/table_route.rs [hunk 4]
```diff
@@ -38,6 +39,7 @@ impl TableRouteKey {
 }
 
 #[derive(Debug, PartialEq, Serialize, Deserialize, Clone)]
+#[serde(tag = "type")]
```
Comment (WenyXu) [inline src/common/meta/src/key/table_route.rs:42]:
```suggestion
#[serde(tag = "type", rename_all = "snake_case")]
```

**Review by fengjiachun — APPROVED** [review]
LGTM

**Pull request merged**